### DISCUSSION TITLE
IP-Adapter style transfer(SDXL)

### DISCUSSION THREAD

**Discussion by Drakosha405** [Acly/krita-ai-diffusion#585] [Ideas]
I try style transfer(SDXL) in IP-Adapter
change ai_diffusion/comfy_workflow.py  314 weight_type="linear", and test it
this is a rough test and maybe if there are better settings for it.
maybe we need it like new options in control layers? and same results with 20% maybe better then 100% but is personal.
 reference | cyberpunk girl | house in forest, lake | dragon in sky
![test](https://github.com/Acly/krita-ai-diffusion/assets/146111805/7fede232-26d9-45e0-9cf0-98d04b42d1cf)

1 row =100%
2 row = 20%
![test1](https://github.com/Acly/krita-ai-diffusion/assets/146111805/302f34e2-5d97-4a31-9fda-c1a23de58aaa)

**Comment by kubilaykilinc** [comment]
Recently, there have been very important improvements in IP Adapter Plus, which is a Comfyui custom node. It would be great to be able to optionally use all the features of Ipadapter Plus and Style transfer and Compositon transfer features in Krita Diffusion.

**Comment by Acly** [comment]
It's merged, I quite like the results. Sadly not a good solution for SD1.5, it sort of works but poor quality.

**Comment by Codeschreibaer** [comment]
I wonder if this example from https://github.com/tencent-ailab/IP-Adapter can already be done with style or / and composition - or eben simpler. Sorry if have a blind spot, and everybody but me knows how to do it. 

![image](https://github.com/Acly/krita-ai-diffusion/assets/71262617/ff1814ff-46f0-4c98-9cad-53ff7c39a736)

**Comment by Codeschreibaer** [comment]
Again: sorry, if "discussions" isnt the right place to talk about my item (don't know). But  i have no idea what compositionan do besides and beyoun style transfer together with style. A short sketch of the range of possibilities independent from eacj other would be nice.

**Reply by Drakosha405** [reply]
[here](https://www.youtube.com/watch?v=czcgJnoDVd4) is a description of the features. yes, we do not have a combined node in Krita and it will be  double the models used.
and I think you're right, it was created just to easily transfer style/composition, no more

**Reply by Acly** [reply]
Don't think it's "double the models used", I tried it out at the time and the combined node produced 100% same results without change in performance.

Technically style transfer uses the basic IPA model (what you get with "Reference") but only applies it to a subset of layers in the diffusion model. Composition works the same, just different layers.
